aom/src/main/java/com/nedap/archie/adlparser/treewalkers/ADLListener.java

```diff
@@ -108,41 +116,65 @@ private void parseArchetypeHRID(TerminalNode hrId) {
 
     public void enterMetaDataItem(AdlParser.MetaDataItemContext ctx) {
         /*
-         SYM_ADL_VERSION '=' VERSION_ID
-        | SYM_UID '=' GUID
-        | SYM_BUILD_UID '=' GUID
-        | SYM_RM_RELEASE '=' VERSION_ID
-        | SYM_IS_CONTROLLED
-        | SYM_IS_GENERATED
         | identifier ( '=' meta_data_value )?
-
          */
-        if(archetype instanceof AuthoredArchetype) {
+        if (archetype instanceof AuthoredArchetype) {
             AuthoredArchetype authoredArchetype = (AuthoredArchetype) archetype;
-
-            if(ctx.metaDataTagAdlVersion() != null) {
-                authoredArchetype.setAdlVersion(ctx.VERSION_ID().getText());
-            }
-            if(ctx.metaDataTagBuildUid() != null) {
-                authoredArchetype.setBuildUid(ctx.GUID().getText());
-            }
-            if(ctx.metaDataTagRmRelease() != null) {
-                authoredArchetype.setRmRelease(ctx.VERSION_ID().getText());
-            }
-            if(ctx.metaDataTagIsControlled() != null) {
-                authoredArchetype.setControlled(true);
-            }
-            if(ctx.metaDataTagIsGenerated() != null) {
-                authoredArchetype.setGenerated(true);
-            }
-            if(ctx.metaDataTagUid() != null) {
-                authoredArchetype.setUid(ctx.GUID().getText());
-            }
-            else if(ctx.identifier() != null) {
-                authoredArchetype.addOtherMetadata(ctx.identifier().getText(), ctx.metaDataValue() == null ? null : ctx.metaDataValue().getText());
+            String identifier = ctx.identifier().getText();
+            String metaDataValue = ctx.metaDataValue() != null ? ctx.metaDataValue().getText() : null;
+
+            // If metaDataValue present, value can be 'primitive_value', 'GUID' or 'VERSION_ID'
+            Pattern versionIdRegex = Pattern.compile("[0-9]+.[0-9]+.[0-9]+((-rc|-alpha)(.[0-9]+)?)?");
```

Comment:
Could you move these calls to a static final constant Now they still get compiled every call, which is quite slow.